data/index.js

```diff
@@ -84,11 +89,7 @@ export function registerSelectors( reducerKey, newSelectors ) {
  */
 export function select( reducerKey ) {
 	if ( arguments.length > 1 ) {
-		// eslint-disable-next-line no-console
-		console.warn(
-			'Deprecated: `select` now accepts only a single argument: the reducer key. ' +
-			'The return value is an object of selector functions.'
-		);
+		deprecated( 'Calling select with multiple arguments', '2.4', undefined, 'Gutenberg' );
```

Comment:
Have you consider using an object with named params instead. It would help to avoid passing `undefined` and make it easier to reason which param does what when reading code.

Reply:
Yes, I was hesitant because the calling side will be more verbose. I don't have a strong opinion though. Happy to update with what y'all think is best.

Reply:
Yes, it'd produce more work for developers :(

Reply:
I agree with passing an object param rather than positional params. WordPress is rife with functions that contain positional parameters which are hard to remember the order of. In some cases a positional parameter has become deprecated requiring that it be change to `$deprecated` in the function signature, and requires always passing `null` in each call. Using an object is more flexible and will allow adding additional parameters in the future as required.